config.go

```diff
@@ -74,12 +76,28 @@ type Config struct {
 	SASLPassword  string
 }
 
-func (c *Config) saslEnabled() bool {
-	return c.SASLUsername != "" && c.SASLPassword != ""
+func (c *Config) useTLS() bool {
+	return c.ClientCert != "" || c.CACert != "" || c.brokerHasCACert()
+}
+
+// brokerHasCACert determines if the broker's certificate has been signed by a CA.
+// Returns `true` if we are confident that the broker uses a CA-signed cert.
+// Returns `false` otherwise, which includes cases where we are not sure (e.g. server offline).
+func (c *Config) brokerHasCACert() bool {
+	conn, err := net.Dial("tcp", c.Servers[0])
+	if err != nil {
+		return false
+	}
+	cfg, err := newTLSConfig("", "", "", true)
+	if err != nil {
+		return false
+	}
```

Comment:
Now that I think of it, this could be a nice thing for the Open() method. If the server is not reachable or if we cannot load the system's cert pool, it makes no sense to even try reading something before that is fixed.

Reply:
@lovromazgon please take a look at this https://github.com/ConduitIO/conduit-connector-kafka/pull/29/commits/f6bb8ee6ae3b042a623d3ebaa6b74bea1e74ad97 when you get to it and let me know what you think of it.: )

Reply:
LGTM, posted just a nitpick about the function naming.